evaluation/red_teaming.py

```diff
@@ -27,48 +30,70 @@ async def run_red_teaming(
     config: dict,
     red_teaming_llm: PromptChatTarget,
     prompt_target: PromptChatTarget,
+    max_turns: int,
+    compare: bool,
 ):
     """Run red teaming attack with provided scorers using Red Teaming Orchestrator."""
+    prompt_target_list = []
+    if compare and isinstance(prompt_target, AppChatTarget):
+        compared_models = config.get("models")
+        for compare_model in compared_models:
+            prompt_target_copy = copy.copy(prompt_target)
+            prompt_target_copy.target_parameters = copy.deepcopy(prompt_target.target_parameters)
+            prompt_target_copy.target_parameters["overrides"]["set_model"] = compare_model
```

Comment:
Why can't we directly create a deep copy of the `prompt_target` object and update the field we need? Doesn't this work?
```suggestion
            prompt_target_copy = copy.deepcopy(prompt_target)
            prompt_target_copy.target_parameters["overrides"]["set_model"] = compare_model
```

Reply:
> Why can't we directly create a deep copy of the `prompt_target` object and update the field we need? Doesn't this work?

I have tried that and it doesn't work because there is a DuckDBPyConnection object inside the prompt_target and it can't be copied directly because they are associated with external resources.

Reply:
Let's have a comment about this then!